chainer/links/loss/hierarchical_softmax.py

```diff
@@ -299,19 +310,50 @@ def __init__(self, in_size, tree):
         # This function object is copied on every forward computation.
         super(BinaryHierarchicalSoftmax, self).__init__()
         self._func = BinaryHierarchicalSoftmaxFunction(tree)
+        self.tree = tree
+        self.init_padding_params()
 
         with self.init_scope():
             self.W = variable.Parameter(uniform.Uniform(1),
                                         (self._func.parser_size, in_size))
 
+    def init_padding_params(self):
+        # Padding `paths` and `codes`
+        paths = self._func.paths
+        codes = self._func.codes
+        begins = self._func.begins
+        n_vocab = self._func.n_vocab
+
+        split_paths = numpy.split(paths, begins[1:-1])
+        split_codes = numpy.split(codes, begins[1:-1])
+
+        max_length = max([len(p) for p in split_paths])
```

Comment:
It consumes too large memory when a very long path exist. I'll fix it after merge.